src/core/passableResult/index.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+
+import { WARN, FAIL } from '../../constants';
+const severities: string[] = [ WARN, FAIL ];
+
+type AsyncObject = null | {
+    [fieldName: string]: {
+        done: boolean,
+        callbacks: Function[]
+    }
+};
+
+const passableResult: Function = (name: string): PassableResult => {
+
+    const completionCallbacks: Function[] = [];
+    let asyncObject: AsyncObject = null;
+
+    /**
+     * Initializes specific field's counters
+     * @param {string} fieldName - The name of the field.
+     */
+    const initFieldCounters: Function = (fieldName: string) => {
+        if (output.testsPerformed[fieldName]) { return output; }
+
+        output.testsPerformed[fieldName] = {
+            testCount: 0,
+            failCount: 0,
+            warnCount: 0
+        };
+    };
+
+    /**
+     * Bumps test counters to indicate tests that's being performed
+     * @param {string} fieldName - The name of the field.
+     */
+    const bumpTestCounter: Function = (fieldName: string) => {
+        if (!output.testsPerformed[fieldName]) { return output; }
+
+        output.testsPerformed[fieldName].testCount++;
+        output.testCount++;
```

Comment:
Not strictly related to this PR, but I feel that if this is a general counter of some sort, it would be cleaner to define it as a getter and compute the total then. Same goes for other total counters.